src/irr/asset/CGLTFLoader.cpp

```diff
@@ -34,16 +35,427 @@ namespace irr
+							return {};
+					}
+
+					auto& generator = value.at_key("generator");
+					if (generator.error() != simdjson::error_code::NO_SUCH_FIELD)
+						header.generator = generator.get_string().value().data();
+
+					auto& copyright = value.at_key("copyright");
+					if (copyright.error() != simdjson::error_code::NO_SUCH_FIELD)
+						header.copyright = copyright.get_string().value().data();
+				}
+
+				/*
+					Buffers and buffer views do not contain type information.
+					They simply define the raw data for retrieval from the file.
+					Objects within the glTF file (meshes, skins, animations) access buffers
+					or buffer views via accessors.
+				*/
+
+				else if (key == "buffers")
+				{
+					for (auto& buffer : value)
+					{
+						auto& byteLength = value.at_key("byteLength");
+						if (byteLength.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							auto byteLengthVal = byteLength.get_uint64().value();
+							if (byteLengthVal < 1)
+								continue;
+						}
+						else
+							continue;
+
+						auto& uri = value.at_key("uri");
+						if (uri.error() != simdjson::error_code::NO_SUCH_FIELD)
+						{
+							std::string_view uriBin = byteLength.get_string().value(); // TODO
+
+							const asset::IAssetLoader::SAssetLoadParams params;
+							auto buffer_bundle = assetManager->getAsset(rootAssetDirectory + uriBin.data(), params);
+							auto buffer = core::smart_refctd_ptr_static_cast<ICPUBuffer>(buffer_bundle.getContents().begin()[0]);
+						}
+						else
+							continue;
+					}
+				}
+
+				else if (key == "bufferViews")
+				{
+					for (auto& bufferView : value)
+					{
+						asset::SBufferBinding<ICPUBuffer> bufferBinding;
+
+						auto& buffer = bufferView.at_key("buffer");
+						if (buffer.error() != simdjson::error_code::NO_SUCH_FIELD)
+							buffer.get_uint64().value(); // TODO
+						else
+							continue;
+					}
+				}
+
+				/*
+					 Meshes are defined as arrays of primitives.
+					 Primitives correspond to the data required for GPU draw calls
+				*/
+
+				else if (key == "meshes")
+				{
+					for (auto& mesh : value)
+					{
+						auto& primitives = mesh.at_key("primitives");
+						if (primitives.error() != simdjson::error_code::NO_SUCH_FIELD)
+							for (auto& primitive : primitives)
+							{
+								auto& attributes = primitive.at_key("attributes");
+								if (attributes.error() != simdjson::error_code::NO_SUCH_FIELD)
+								{
+									auto& position = attributes.at_key("POSITION");
+									auto& normal = attributes.at_key("NORMAL");
+									auto& tangent = attributes.at_key("TANGENT");
+									auto& texcoord0 = attributes.at_key("TEXCOORD_0");
+									auto& texcoord1 = attributes.at_key("TEXCOORD_1");
+									auto& color0 = attributes.at_key("COLOR_0");
+									auto& joint0 = attributes.at_key("JOINTS_0");
+									auto& weight0 = attributes.at_key("WEIGHTS_0");
+
+									// TODO
+								}
+								else
+									continue;
+							}
```

Comment:
wait, it actually goes 1-by-1 to get the primitives!?

Reply:
I will bypass it

Reply:
by changing my API